bundles/org.dataflowanalysis.analysis.dfd/src/org/dataflowanalysis/analysis/dfd/core/DFDCyclicTransposeFlowGraphFinder.java

```diff
@@ -0,0 +1,233 @@
+package org.dataflowanalysis.analysis.dfd.core;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.IdentityHashMap;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.log4j.Logger;
+import org.dataflowanalysis.analysis.core.AbstractTransposeFlowGraph;
+import org.dataflowanalysis.analysis.dfd.resource.DFDResourceProvider;
+import org.dataflowanalysis.dfd.datadictionary.DataDictionary;
+import org.dataflowanalysis.dfd.datadictionary.Pin;
+import org.dataflowanalysis.dfd.dataflowdiagram.DataFlowDiagram;
+import org.dataflowanalysis.dfd.dataflowdiagram.Flow;
+import org.dataflowanalysis.dfd.dataflowdiagram.Node;
+import org.eclipse.emf.ecore.util.EcoreUtil;
+
+/**
+ * The DFDTransposeFlowGraphFinder determines all transpose flow graphs contained in a model, allowing cycles
+ */
+public class DFDCyclicTransposeFlowGraphFinder extends DFDTransposeFlowGraphFinder {
+    private final Logger logger = Logger.getLogger(DFDCyclicTransposeFlowGraphFinder.class);
+    private boolean hasCycles = false;
```

Comment:
Move the assignments of `hasCycles` and `numOfCopies` to the constructor, please